### ISSUE TITLE
Insert only returns identifier not the properties

### ISSUE THREAD

**Issue by robinvriens** [Vincit/objection.js#2245]
I'm trying to insert something in the database as following:

```js
const test = async () => {
  const user = await User.query().insert({
    email: 'test@gmail.com',
    password: 'test123'
  })

  console.log(user)
}
```

On the documentation it says it will return the identifier as well as the properties that are given (in this case that would be the id, email and password. so the expected output for me would be:
```js
User {
  id: '1410336a-1420-4729-9470-3e1c8b771137',
  email: 'test@gmail.com',
  password: 'test123'
}
```
But for some reason if I console log the user it only returns the id and not the other properties as following:
```js
User {
  id: '1410336a-1420-4729-9470-3e1c8b771137',
  email: undefined,
  password: undefined
}
```
Here is my User Model
```js
export default class User extends Model {
  id!: string
  email!: string
  password!: string

  static tableName = 'user'
}
```
Whenever i remove the properties it seems to work but this seems odd, what am I doing wrong?

**Comment by comp615** [comment]
Also seeing this behavior, as mentioned by OP, if I remove the class props, the method correctly returns with values.

**Comment by lehni** [comment]
This sounds like it is the same problem as outlined here:

https://github.com/Vincit/objection.js/issues/2105#issue-994104199

**Comment by lehni** [comment]
Closing as duplicate.

**Comment by lehni** [comment]
Fixed in #2400

**Comment by mrsarm** [comment]
If you ended up here like me because having this issue upgrading Typescript, but you cannot upgrade Objection, and as mention in the other ticket that duplicates this, it has to do with the TS configuration `useDefineForClassFields` being set to `true`, like me you did not touch it in the `tsconfig.json` file, but upgrading from TypeScript 4.x to 5.x indeed the setting is defaulted to `true`, so you have to set it to `false` to keep things working as it was before:

```js
{
  "compilerOptions": {
    "target": "esnext",
    "useDefineForClassFields": false,
    ...
  }
}
```
